apiserver/common/watch.go

```diff
@@ -192,3 +193,61 @@ func (w *MultiNotifyWatcher) Err() error {
 func (w *MultiNotifyWatcher) Changes() <-chan struct{} {
 	return w.changes
 }
+
+// StringsNotifyWatcher wraps a corewatcher.StringsWatcher and
+// provides a NotifyWatcher interface.
+type StringsNotifyWatcher struct {
+	tomb    tomb.Tomb
```

Comment:
Shouldn't we use a catacomb and add the StringsWatcher to it?

Reply:
I'll check.

Reply:
So essentially, we have some ad-hoc catacomb in the notify watcher. I'll make a new PR to fix it.

Reply:
See: https://github.com/juju/juju/pull/17122